src/networking.c

```diff
@@ -438,18 +438,22 @@ static int read_client_data(conn_info *conn) {
  * of what to do.
  */
 static void handle_udp_message(aeEventLoop *loop, int fd, void *edata, int mask) {
-		statsite_networking *netconf = (statsite_networking *) edata;
+    struct iovec vectors[2];
+    int num_vectors;
+    ssize_t read_bytes;
+    statsite_networking *netconf = (statsite_networking *) edata;
 
-    while (1) {
-        // Get the associated connection struct
-        conn_info *conn = netconf->udp_client;
+    // Get the associated connection struct
+    conn_info *conn = netconf->udp_client;
 
-        // Clear the input buffer
-        circbuf_clear(&conn->input);
+    // Clear the input buffer
+    circbuf_clear(&conn->input);
 
+    // Loop until our buffer cannot hold another UDP packet (using a 1500
+    // byte MTU) or we cannot read another packet off the wire.  We do not
+    // read continuously off of the UDP socket to preserve timer execution.
+    while (circbuf_avail_buf(&conn->input) > 1500) {
```

Comment:
I had similar thoughts, but wasn't sure where to find/build a better value.  UDP packet size is a fairly well understood StatsD limitation, so I'm leaning toward just documenting it.